tests/functional/cylc-clean/00-basic.t

```diff
@@ -62,23 +67,23 @@ cmp_ok "${TEST_NAME}.stdout" << __TREE__
 ${TEST_DIR}/sym-cycle
 \`-- cylc-run
     \`-- cylctb-${CYLC_TEST_TIME_INIT}
-        \`-- f
+        \`-- ${FUNCTIONAL_DIR}
```

Comment:
Just realised that if another test gets added to this directory (i.e. `01-<name>.t`) then this test will break.

Reply:
I tried copying another test file into `tests/functional/cylc-clean/01-something.t` and `00-basic.t` still works

Reply:
If 01-something creates a workflow and the two tests are run together then 01-something will change the output of `tree` based on the order in which the tests are run.

Reply:
I tried making a copy of `00-basic.t` called `01-foo.t` that has a slightly different expected tree (no symlink `work` dir or `leave-me-alone` dir) and both pass? Doesn't `purge` take care of removing the old stuff?

Reply:
It does, but only once the test has passed but tests run in parallel.

Reply:
Hopefully https://github.com/cylc/cylc-flow/pull/3961/commits/b35b10ba000f18dac5303b9bcd5393ceb1a2c6ed prevents that